src/roman.rs

```diff
@@ -91,25 +91,27 @@ impl Roman {
     /// cannot be appropriately formatted using the seven standard numerals.
     pub fn new(n: u16) -> Option<Roman> {
         match n {
-            n @ 1..=4999 => Some(Roman(n)),
+            n if n <= 4999 => NonZeroU16::new(n).map(Roman),
             _ => None,
         }
     }
 
-    /// Creates a `Roman` value based on a `u16`.
+    /// Creates a `Roman` value based on a [`NonZeroU16`].
```

Comment:
I didn't know you could make links this way. I assume that's what this is. It's neat.

Reply:
It's a feature that was recently [stabilized](https://blog.rust-lang.org/inside-rust/2020/09/17/stabilizing-intra-doc-links.html), though everyone was using it even before the stabilization.